renku/domain_model/dataset.py

```diff
@@ -652,6 +652,13 @@ def update_metadata_from(self, other: "Dataset", exclude=None):
         if self.date_published is not None:
             self.date_created = None
 
+        # NOTE: Fix image IDs, in some cases the image IDs set by the providers can be malformed
+        # and not match the SHACL definition for Renku. This cannot be addressed in the dataset
+        # providers because the dataset providers do not have access to the dataset ID which is needed
+        # for setting the dataset image ID.
+        for image_ind in range(len(self.images)):
+            self.images[image_ind].id = f"{self.id}/images/{image_ind}"
```

Comment:
ImageObject has a `generate_id(dataset_id, position)` method and a `position` field that should be used here

Reply:
Awesome. I will use that then.